src/sacn/receiver.c

```diff
@@ -774,24 +786,35 @@ etcpal_error_t assign_receiver_to_thread(SacnReceiver* receiver, const SacnRecei
 
   SACN_ASSERT(assigned_thread);
 
-  etcpal_error_t res = sacn_add_receiver_socket(receiver->thread_id, kEtcPalIpTypeV4, config->universe_id,
-                                                receiver->netints, receiver->num_netints, &receiver->ipv4_socket);
-  if (res == kEtcPalErrNoNetints)
-    res = kEtcPalErrOk;  // Try IPv6.
+  etcpal_error_t res = kEtcPalErrOk;
+
+  if ((receiver->ip_supported == kSacnIpV4Only) || (receiver->ip_supported == kSacnIpV4AndIpV6))
+  {
+    res = sacn_add_receiver_socket(receiver->thread_id, kEtcPalIpTypeV4, config->universe_id, receiver->netints,
+                                   receiver->num_netints, &receiver->ipv4_socket);
+
+    if ((res == kEtcPalErrNoNetints) && (receiver->ip_supported == kSacnIpV4AndIpV6))
+      res = kEtcPalErrOk;  // Try IPv6.
+  }
 
   if (res == kEtcPalErrOk)
   {
-    res = sacn_add_receiver_socket(receiver->thread_id, kEtcPalIpTypeV6, config->universe_id, receiver->netints,
-                                   receiver->num_netints, &receiver->ipv6_socket);
+    if ((receiver->ip_supported == kSacnIpV6Only) || (receiver->ip_supported == kSacnIpV4AndIpV6))
+    {
+      res = sacn_add_receiver_socket(receiver->thread_id, kEtcPalIpTypeV6, config->universe_id, receiver->netints,
+                                     receiver->num_netints, &receiver->ipv6_socket);
+    }
   }
 
   if (res == kEtcPalErrOk && !assigned_thread->running)
   {
     res = start_receiver_thread(assigned_thread);
     if (res != kEtcPalErrOk)
     {
-      sacn_remove_receiver_socket(receiver->thread_id, receiver->ipv4_socket, true);
-      sacn_remove_receiver_socket(receiver->thread_id, receiver->ipv6_socket, true);
+      if (receiver->ipv4_socket != ETCPAL_SOCKET_INVALID)
```

Comment:
Likewise, you are doing this same cleanup in a spots, it could be a utility function.